frontend/src/components/elements/BokehChart/BokehChart.tsx

```diff
@@ -15,54 +15,62 @@
  * limitations under the License.
  */
 
-import React, { PureComponent, ReactNode } from "react"
+import React, { ReactNode, useEffect } from "react"
 import { Map as ImmutableMap } from "immutable"
 import { embed as BokehEmbed } from "@bokeh/bokehjs"
 import withFullScreenWrapper from "hocs/withFullScreenWrapper"
 
-interface Props {
+export interface BokehChartProps {
   width: number
   element: ImmutableMap<string, any>
   index: number
-}
-
-export interface PropsWithHeight extends Props {
-  height: number | undefined
+  height?: number
 }
 
 interface Dimensions {
   width: number
   height: number
 }
 
-export class BokehChart extends PureComponent<PropsWithHeight> {
-  private chartId = `bokeh-chart-${this.props.index}`
+export function BokehChart({
+  width: w,
+  element,
+  index,
+  height: h,
+}: BokehChartProps): ReactNode {
+  const chartId = `bokeh-chart-${index}`
 
-  private getChartData = (): any => {
-    const figure = this.props.element.get("figure")
+  const getChartData = (): any => {
+    const figure = element.get("figure")
     return JSON.parse(figure)
   }
 
-  public getChartDimensions = (plot: any): Dimensions => {
-    const useContainerWidth = this.props.element.get("useContainerWidth")
+  const getChartDimensions = (plot: any): Dimensions => {
+    const useContainerWidth = element.get("useContainerWidth")
     // Default values
-    let width = plot.attributes.plot_width
-    let height = plot.attributes.plot_height
+    let width: number = plot.attributes.plot_width
+    let height: number = plot.attributes.plot_height
 
     // if is not fullscreen and useContainerWidth==false, we should use default values
-    if (this.props.height) {
+    if (h) {
```

Comment:
Hmmm, I'm not a fan of one letter variables. Couple choices:
* naming props like `propWidth` and `propHeight`
* defining `width` and `height` from `getChartDimensions` to be `chartWidth` and `chartHeight` (My preference as that doesn't impact the expectation of the props and keeps clarity)

Reply:
Yes, agreed. I will use chartWidth and chartHeight